crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -277,7 +277,7 @@ impl ShaderCache {
         data.pipelines.insert(pipeline);
 
         // PERF: this shader_defs clone isn't great. use raw_entry_mut when it stabilizes
-        let module = match data.processed_shaders.entry(shader_defs.to_vec()) {
+        let module = match data.processed_shaders.entry(shader_defs.into()) {
```

Comment:
```suggestion
        let module = match data.processed_shaders.entry_ref(&shader_defs) {
```
Might be able to avoid allocation here entirely via `entry_ref`.